packages/orchestrator-ui-components/src/components/WfoForms/formFields/ContactPersonNameField.tsx

```diff
@@ -0,0 +1,253 @@
+import { useAxiosApiClient } from '../useAxiosApiClient';
+
+export function stop(e: React.SyntheticEvent) {
+    if (e !== undefined && e !== null) {
+        e.preventDefault();
+        e.stopPropagation();
+    }
+}
+
+export type ContactPersonNameFieldProps = FieldProps<
+    string,
+    { organisationId?: string; organisationKey?: string }
+>;
+
+declare module 'uniforms' {
+    interface FilterDOMProps {
+        organisationId: never;
+        organisationKey: never;
+    }
+}
+filterDOMProps.register('organisationId', 'organisationKey');
+
+function ContactPersonName({
+    disabled,
+    id,
+    inputRef = React.createRef() as Ref<HTMLInputElement>,
+    label,
+    description,
+    name,
+    onChange,
+    placeholder,
+    readOnly,
+    value,
+    error,
+    showInlineError,
+    errorMessage,
+    organisationId,
+    organisationKey,
+    ...props
+}: ContactPersonNameFieldProps) {
+    const apiClient = useAxiosApiClient();
+    const t = useTranslations('forms');
+    const { model, onChange: formOnChange, schema } = useForm();
+
+    const contactsPersonFieldNameArray = joinName(null, name).slice(0, -1);
+    const emailFieldName = joinName(contactsPersonFieldNameArray, 'email');
+    const contactsFieldName = joinName(
+        contactsPersonFieldNameArray.slice(0, -1),
+    );
+
+    const chosenPersons: ContactPerson[] = get(model, contactsFieldName, []);
+    // We cant call useField conditionally so if we don't have a parent we call it for ourself
+    const useFieldName = contactsPersonFieldNameArray.length
+        ? contactsFieldName
+        : name;
+    const contactsField = useField(useFieldName, {}, { absoluteName: true })[0];
+
+    const organisationFieldName =
+        organisationKey ||
+        contactsField.field.organisationKey ||
+        'organisation';
+
+    // Get initial value for org field if it exists (we cant really test)
+    let organisationInitialValue;
+    try {
+        organisationInitialValue = schema.getInitialValue(
+            organisationFieldName,
+            {},
+        );
+    } catch {
+        organisationInitialValue = '';
+    }
+
+    const organisationIdValue =
+        organisationId ||
+        contactsField.field.organisationId ||
+        get(model, organisationFieldName, organisationInitialValue);
+
+    const [displayAutocomplete, setDisplayAutocomplete] = useState(false);
+    const [contactPersons, setContactPersons] = useState<ContactPerson[]>([]);
+    const [selectedIndex, setSelectedIndex] = useState(-1);
+
+    const suggestions = value
+        ? contactPersons
+              .filter(
+                  (item) =>
+                      item.name.toLowerCase().indexOf(value.toLowerCase()) > -1,
+              )
+              .filter(
+                  (item) =>
+                      !chosenPersons.some(
+                          (person) => person.email === item.email,
+                      ),
+              )
+        : [];
+
+    useEffect(() => {
```

Comment:
Maybe you can use react-query here ?

Reply:
There are a few ways we get data now and that's a good point. Let's make that one of the points to discuss Thursday. I'm going to leave it like this now